src/Analyzers/CSharp/Analyzers/UseUTF8StringLiteral/UseUTF8StringLiteralDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Immutable;
+using System.Data;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseUTF8StringLiteral
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed partial class UseUTF8StringLiteralDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public const string StringValuePropertyName = "StringValue";
+
+        public UseUTF8StringLiteralDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseUTF8StringLiteralDiagnosticId,
+                EnforceOnBuildValues.UseUTF8StringLiteral,
+                CSharpCodeStyleOptions.PreferUTF8StringLiteral,
+                LanguageNames.CSharp,
+                new LocalizableResourceString(nameof(CSharpAnalyzersResources.Convert_to_UTF8_string_literal), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_UTF8_string_literal), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.LanguageVersion() < LanguageVersion.Preview)
```

Comment:
@jcouv for the preferred pattern for doing preview checks.  i think we end up defining a CSharp11 (or 12) constant somewhere in the IDE and compare to that.  then we just chnage that one location to point at the actual compielr version once available.

Reply:
In the compiler layer we use an alias "Next" which means "11". It's not part of `LanguageVersion` enum since we didn't want to affect the public API, but it's in `TestOptions.RegularNext` and `LanguageVersionFacts.CSharpNext`. Those are just indirections for "Preview" at the moment. Before shipping C# 11, we'll replace all those references with "Next"=>"11".

For the IDE layer, I would suggest doing the same: declare an indirection/alias for what will be C# 11. That gives a central place to fix once LangVer 11 is introduced. 

If you do that, please add a mention in comment on `RegularNext` so that we remember to adjust it.

Reply:
The IDE should already have this available:

https://github.com/dotnet/roslyn/blob/6023476974b6348c6660a1e64e183e2b2e46dde0/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/LanguageVersionExtensions.cs#L18